### PR TITLE
Fix a few compiler warnings

### PR DESCRIPTION
**Pull request by daboehme**

### CODE REVIEW

File: src/gotcha_auxv.c [hunk 1]
```diff
@@ -227,7 +227,7 @@ struct link_map *get_vdso_from_maps()
    struct link_map *m;
    maps = gotcha_open("/proc/self/maps", O_RDONLY);
    for (;;) {
-      hit_eof = read_line(line, 4097, maps);
+      hit_eof = read_line(line, 4096, maps);
```
Comment (hariharan-devarajan) [inline src/gotcha_auxv.c:230]: This looks fine to me. Just that, to avoid this kind of bug, it might be better to define a constant?

**Review by hariharan-devarajan — CHANGES_REQUESTED** [review]

File: src/gotcha_auxv.c (hunk 1 above)
Comment (daboehme) [reply src/gotcha_auxv.c:230]: The read_line function writes a zero-byte to size-1, i.e. line[4096] which is out-of-bounds. So it is a bug.

Comment (hariharan-devarajan) [reply src/gotcha_auxv.c:230]: Oh i meant, your fix looks good. I was saying, we should change the line array as well to use a constant.

```
#define BUFFER_LEN 4096
char name[BUFFER_LEN], line[BUFFER_LEN], *line_pos;
   struct link_map *m;
   maps = gotcha_open("/proc/self/maps", O_RDONLY);
   for (;;) {
      hit_eof = read_line(line, BUFFER_LEN, maps);
```

As I see this buffer length used in more places in the file like ```parse_auxv_contents()```

Comment (daboehme) [reply src/gotcha_auxv.c:230]: Oh, OK. Yeah I guess it makes sense to use a constant.

Comment (hariharan-devarajan) [reply src/gotcha_auxv.c]: I did the fix.

**Review by hariharan-devarajan — APPROVED** [review]
The changes look good.

File: src/gotcha_auxv.c [hunk 2]
```diff
@@ -31,14 +31,14 @@ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
...
+#define BUFFER_LEN 4096
 
 int parse_auxv_contents()
 {
    char name[] = "/proc/self/auxv";
    int fd, done = 0;
-   char buffer[4096];
-   ssize_t buffer_size = 4096, offset = 0, result;
+   char buffer[BUFFER_LEN];
+   ssize_t buffer_size = BUFFER_LEN, offset = 0, result;
```
Comment (bertwesarg) [inline src/gotcha_auxv.c:41]: `buffer_size` is basically `const`. so the variable should either be declared as this or replaced by `BUFFER_LEN`

Comment (hariharan-devarajan) [reply src/gotcha_auxv.c:41]: Yes thats why we added ```#define BUFFER_LEN 4096``` at the top.

Comment (hariharan-devarajan) [reply src/gotcha_auxv.c]: I changed the buffer_size to be a const.

**Pull request merged**